examples/ffa_simulation.rs

```diff
@@ -0,0 +1,230 @@
+    let dy = (y0 - y1).abs();
+    return dx + dy;
+}
+
+// Spawns entities one by one. It's preferable to use spawn_batch instead, which has better performance for spawning multiple entities.
+fn _spawn_enitites(world: &mut World, n: usize) {
+    let mut rng = thread_rng();
+
+    for _ in 0..n {
+        let pos = Position {
+            x: rng.gen_range(-10, 10),
+            y: rng.gen_range(-10, 10),
+        };
+        let s = Speed {
+            speed: rng.gen_range(1, 5),
+        };
+        let hp = Health {
+            hp: rng.gen_range(30, 50),
+        };
+        let dmg = Damage {
+            dmg: rng.gen_range(1, 10),
+        };
+        let kc = KillCount { count: 0 };
+
+        world.spawn((pos, s, hp, dmg, kc));
+    }
+}
+
+fn batch_spawn_entities(world: &mut World, n: usize) {
+    let mut rng = thread_rng();
+
+    let to_spawn = (0..n).map(|_| {
+        let pos = Position {
+            x: rng.gen_range(-10, 10),
+            y: rng.gen_range(-10, 10),
+        };
+        let s = Speed {
+            speed: rng.gen_range(1, 5),
+        };
+        let hp = Health {
+            hp: rng.gen_range(30, 50),
+        };
+        let dmg = Damage {
+            dmg: rng.gen_range(1, 10),
+        };
+        let kc = KillCount { count: 0 };
+
+        (pos, s, hp, dmg, kc)
+    });
+
+    world.spawn_batch(to_spawn);
+}
+
+fn system_integrate_motion(world: &mut World) {
+    let mut rng = thread_rng();
+
+    for (id, (pos, s)) in &mut world.query::<(&mut Position, &Speed)>() {
+        let change = (
+            rng.gen_range(-s.speed, s.speed),
+            rng.gen_range(-s.speed, s.speed),
+        );
+        pos.x += change.0;
+        pos.y += change.1;
+        println!("Unit {:?} moved to {:?}", id, pos);
+    }
+}
+
+// In this system entities find the closest entity and fire at them
+fn system_fire_at_closest(world: &mut World) {
+    for (id0, (pos0, dmg0, kc0)) in
+        &mut world.query::<With<Health, (&Position, &Damage, &mut KillCount)>>()
+    {
+        let mut min_dist: Option<i32> = None;
+        let mut closest_id: Option<Entity> = None;
+
+        //Find closest:
+        //Comment about implementation: Nested queries are O(n^2) and you usually want to avoid that by using some sort of spatial index like a quadtree or more general BVH, which we don't bother with here since it's out of scope for the example.
+        'child_loop: for (id1, pos1) in &mut world.query::<With<Health, &Position>>() {
+            if id0 == id1 {
+                continue 'child_loop;
+            }
+
+            let dist = manhattan_dist(pos0.x, pos1.x, pos0.y, pos1.y);
+
+            match min_dist {
+                None => {
+                    min_dist = Some(dist);
+                }
+                Some(mut _dist0) => {
+                    _dist0 = _dist0.min(dist);
+                }
+            }
+
+            if Some(dist) == min_dist {
+                closest_id = Some(id1);
+            }
+        }
```

Comment:
I think there is a simpler way which avoids mutation and unwrapping, Not sure if it is more clearer though

```rust
for (id0, (pos0, dmg0, kc0)) in &mut world.query::<With<Health, (&Position, &Damage, &mut KillCount)>>() {
    // find closest:
    let mut entities = world.query::<With<Health, &Position>>();
    let closest = entities.iter()
        .filter(|(id1, _)| *id1 != id0)
        .min_by_key(|(_, pos1)| manhattan_dist(pos0.x, pos1.x, pos0.y, pos1.y));

     match closest {
         Some(closest) => {
             let (id1, _) = closest;
             // health code
         }
         None => {
             println!("{:?} is the last survivor!", id0);
         }
}
```